lib/pwned_password_downloader.rb

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/env ruby
+require 'fileutils'
+require 'net/http/persistent'
+require 'ruby-progressbar'
+
+class PwnedPasswordDownload
+  attr_reader :destination,
+              :num_threads,
+              :keep_threshold
+
+  def initialize(
+    destination: 'tmp/pwned',
+    num_threads: 64,
+    keep_threshold: 30
+  )
+    @destination = destination
+    @num_threads = num_threads
+    @keep_threshold = keep_threshold
+  end
+
+  def run!
+    queue = Queue.new
+
+    ('00000'.to_i(16)..'FFFFF'.to_i(16)).each do |prefix_num|
```

Comment:
for ease of testing i'd move these to be parameterized
```suggestion
  def run!(start: '00000', finish: 'FFFFF')
    queue = Queue.new

    (start.to_i(16)..finish.to_i(16)).each do |prefix_num|
```

Reply:
> for ease of testing i'd move these to be parameterized

Updated to parameterize in e596c9c16.